test/cypress/integration/collection_upload.js

```diff
@@ -0,0 +1,31 @@
+describe('Collection Upload Tests', () => {
+  var adminUsername = Cypress.env('username');
+  var adminPassword = Cypress.env('password');
+
+  beforeEach(() => {
+    cy.login(adminUsername, adminPassword);
+  });
+
+  it('collection is uploaded', () => {
+    cy.intercept(
+      'GET',
+      Cypress.env('prefix') + '_ui/v1/collection-versions/?namespace=*',
+    ).as('upload');
+    const filepath = 'collections/ansible-network-1.2.0.tar.gz';
+    cy.galaxykit('-i namespace create', 'ansible');
+    cy.menuGo('Collections > Namespaces');
+    cy.intercept('GET', Cypress.env('prefix') + '_ui/v1/repo/published/*').as(
+      'namespaces',
+    );
+
+    cy.get('a[href="/ui/repo/published/ansible"]').click();
+    cy.wait('@namespaces');
+    cy.contains('Upload collection').click();
+    cy.get('input[type="file"]').attachFile(filepath);
```

Comment:
```suggestion
    // test/cypress/fixtures/collections/ansible-network-1.2.0.tar.gz
    cy.fixture('collections/ansible-network-1.2.0.tar.gz', 'binary')
      .then(Cypress.Blob.binaryStringToBlob)
      .then(fileContent => {
        cy.get('input[type="file"]').attachFile({
          fileContent,
          fileName: 'ansible-network-1.2.0.tar.gz',
          mimeType: 'application/gzip',
        });
      });
```

(as per https://www.npmjs.com/package/cypress-file-upload#working-with-raw-file-contents)

Reply:
(this works, but sometimes waiting for "Completed" fails because the task takes more than 4 seconds)